### ISSUE TITLE
Setup Package Publishing

### ISSUE THREAD

**Issue by seveibar** [eelab-dev/EEcircuit#23]
Hey @danchitnis , wondering if you would publish a module that publishes the ngspice bindings as a standalone javascript/wasm module.

So somebody could do e.g.

```bash
npm add eecircuit
```

I'm not 100% sure on the API, but something like this?

```ts
import { SpiceSimulation } from "eecircuit"

const spiceSim = new SpiceSimulation()

spiceSim.setSpice(`
voltage divider netlist
V1 in 0 1
R1 in out 1k
R2 out 0 2k
.end
`)

await spiceSim.run()
```

I organized the API such that it could maybe stream responses eventually which i think ngspice is capable of.

I would love to help contribute this if it's interesting to you. My biggest concern (something I would need to report back on) is the loader. Runtimes like `bun` and `deno` can directly import WASM but IIRC it's not that simple with other runtimes.

If this isn't interesting and you'd like this repo to stay at the application level that's ok as well!

**Comment by danchitnis** [comment]
Do you mean for use in node/deno or web/bundler? The issue is ngspice is not made for this environment, so it is a bit of a hack to keep it running efficiently. It can probably be converted into a class with a constructor. I need to check the code and see if this is possible. Also, is uploading WASM binaries to npm a thing?

**Comment by seveibar** [comment]
@danchitnis yep you can upload WASM files to npm! I would love to help with this, I could try to help PoC! Just wanted to make sure I wasn't stepping on your toes if it's something you wanted to do!!!

sidenote: I think the tscircuit schematics are getting quite good and I think would look cool on eecircuit!! Maybe I could try to convert spice into schematics automatically or something- i think that's a hard project and maybe for later 😁

**Comment by danchitnis** [comment]
Yes, I am happy with that direction. Let me do a few tests and get back to you.
